atlite/convert.py

```diff
@@ -493,6 +493,76 @@ def wind(cutout, turbine, smooth=False, **params):
     )
 
 
+# irradiation
+def convert_irradiation(
+    ds, orientation, irradiation="total", trigon_model="simple", clearsky_model="simple"
+):
+    solar_position = SolarPosition(ds)
+    surface_orientation = SurfaceOrientation(ds, solar_position, orientation)
+    irradiation = TiltedIrradiation(
+        ds,
+        solar_position,
+        surface_orientation,
+        trigon_model=trigon_model,
+        clearsky_model=clearsky_model,
+        irradiation=irradiation,
+    )
+    return irradiation
+
+
+def irradiation(
+    cutout, orientation, irradiation="total", clearsky_model=None, **params
+):
+    """
+    Calculate the total, direct, diffuse, or ground irradiation on a tilted
+    surface.
+
+    Parameters
+    ----------
+    orientation : str, dict or callback
+        Surface orientation can be chosen from either
+        'latitude_optimal', a constant orientation {'slope': 0.0,
+        'azimuth': 0.0} or a callback function with the same signature
+        as the callbacks generated by the
+        'atlite.pv.orientation.make_*' functions.
+    irradiation : str
+        The irradiation quantity to be returned. Defaults to "total" for total
+        irradiation. Other options include "direct" for direct irradiation,
+        "diffuse" for diffuse irradation, and "ground" for ground irradiation.
+    clearsky_model : str or None
+        Either the 'simple' or the 'enhanced' Reindl clearsky
+        model. The default choice of None will choose dependending on
+        data availability, since the 'enhanced' model also
+        incorporates ambient air temperature and relative humidity.
```

Comment:
I know that this is what the `.pv(..)` documentation says; which by itself isn't very comprehensive:

The `clearsky_model` is only applied for determining shares of direct and diffuse irradiation *if* they are not already provided by the `cutout` data. I.e. if you use `ERA5` or `SARAH` data, the `clearsky_model` parameter doesn't do anything and no realsky model is applied. (Because cutouts from both datasets include `influx_direct` and `influx_diffuse` variables and thus no calculation happens in `TiltedIrradiation(...)`). This is part of a larger portion of legacy code which was required in the past when we were supporting datasets with only `influx` variables.

Maybe that's also relevant for your implementation?

Reply:
Shouldn't be relevant as far as I can tell. My implementation basically just needs access to the raw irradiation data already used by the PV functions, but without the radiation-to-generation conversion.

Reply:
Added a clarification in the docsrting.